### ISSUE TITLE
agentaddress udp:161 with linux VRF support (iface binding) is broken

### ISSUE THREAD

**Issue by stannous** [net-snmp/net-snmp#4]
At some point after linux VRF support for all interfaces (udp:161) was working.
Now, with `agentaddress udp:161@red`, we parse this incorrectly and listen on
> udp        0      0 0.0.0.161:161           0.0.0.0:* 

Regular IP addresses bound to a VRF interface work correctly
`agentaddress 12.0.0.10@red`

> udp        0      0 12.0.0.10:161           0.0.0.0:*                          

Thanks,
Sam

**Comment by fenner** [comment]
Hi Sam,

The @ is attached to the IP address, so when you write udp:161@red it
parses 161 as the IP address (and then uses the default 161 port).  Try
udp:@red:161?

  Bill

**Comment by stannous** [comment]
Thanks, Bill.  That works.

I also tried "udp:@red:161" along with something in the default VRF
"11.0.0.10" but it complains about the latter.  Also, the combination of "udp:161" and
"12.0.0.10@red" does not work.  I think this is the way it has always worked (since VRFs
were added) but I can't remember.   So it appears that we can't have anything else with
an "all".  Which is fine.

Multiple IP addresses in different VRFs still works.

Thanks,
Sam
